common/src/main/java/com/wynntils/core/net/UrlManager.java

```diff
@@ -342,4 +528,18 @@ private static final class UrlProfile {
         String md5;
         String encoding;
     }
+
+    private record UrlList(int version, Map<UrlId, UrlInfo> urls) {
```

Comment:
This name confused the heck out of me. It's not a list? Perhaps use `VersionedUrlMap`? Or just `UrlMap`?

Reply:
Ok, I think I understand now. We would need a clear name on what the file "urls.json" represents. It is not a "url", it is a set of mapping urlid->url, a version, and a provenance. 

Maybe "URLs holder" (`UrlsHolder`)? Getting the plural -s might help saying that we are dealing with a collection of urls.

Reply:
Urgh. I dunno, really. UrlsHolder does not sound good, either. UrlCollection? UrlMapper? Hm, yeah, that might work. Then `readUrls` would be `readUrlMapper`, and `urlListMap` would be `urlMappers`. I think that would be good.